relay/relay.go

```diff
@@ -273,11 +275,34 @@ func (r *Relay) process(parentCtx context.Context) error {
 		}
 	}()
 
+	readerRetry, err := retry.NewReaderRetry(r.cfg.ReaderRetry)
+	if err != nil {
+		return err
+	}
+
 	transformer2 := transformer.NewTransformer(parser2)
 
 	go r.doIntervalOps(parentCtx)
 
-	return r.handleEvents(parentCtx, reader2, transformer2, writer2)
+	// handles binlog events with retry mechanism.
+	// it only do the retry for some binlog reader error now.
+	for {
+		err := r.handleEvents(parentCtx, reader2, transformer2, writer2)
+		if !readerRetry.Check(parentCtx, err) {
```

Comment:
check whether `err` is nil before `readerRetry.Check`, and return directly if `err` is nil?

Reply:
if `err` is `nil`, I think we should return directly (by cancel)

Reply:
return directly if `err` is `nil` in https://github.com/pingcap/dm/pull/265/commits/6f8de211a7a8435c473053577f898db21e495489.